src/main/java/org/openstreetmap/atlas/checks/database/FlagDatabaseSubCommand.java

```diff
@@ -0,0 +1,376 @@
+    private static final String FLAG_PATH_INPUT = "flag_path";
+    private static final String DATABASE_URL_INPUT = "database_url";
+    private static final String LOG_EXTENSION = "log";
+    private static final String ZIPPED_LOG_EXTENSION = ".log.gz";
+    private static final String ISO_COUNTRY_CODE = "iso_country_code";
+    private static final String OSM_ID_LEGACY = "osmid";
+    private static final int THREE = 3;
+    private static final int FOUR = 4;
+    private static final int FIVE = 5;
+    private static final int SIX = 6;
+    private static final int SEVEN = 7;
+    private static final int EIGHT = 8;
+    private static final int BATCH_SIZE = 1000;
+
+    /**
+     * An enum containing the different types of input files that we can handle.
+     */
+    private enum OutputFileType
+    {
+        LOG,
+        COMPRESSED_LOG
+    }
+
+    private static final Gson gson = new GsonBuilder()
+            .registerTypeAdapter(CheckFlag.class, new CheckFlagDeserializer()).create();
+    private static final Logger logger = LoggerFactory.getLogger(FlagDatabaseSubCommand.class);
+    private static final Set<String> blacklistKeys = new HashSet<>();
+
+    static
+    {
+        blacklistKeys.add("itemType");
+        blacklistKeys.add("identifier");
+        blacklistKeys.add(OSM_ID_LEGACY);
+        blacklistKeys.add("osmIdentifier");
+        blacklistKeys.add("relations");
+        blacklistKeys.add("members");
+        blacklistKeys.add(ISO_COUNTRY_CODE);
+    }
+
+    private final OptionAndArgumentDelegate optionAndArgumentDelegate;
+    private Timestamp timestamp;
+
+    public static void main(final String[] args)
+    {
+        new FlagDatabaseSubCommand().runSubcommandAndExit(args);
+    }
+
+    public FlagDatabaseSubCommand()
+    {
+        this.optionAndArgumentDelegate = this.getOptionAndArgumentDelegate();
+    }
+
+    @Override
+    @SuppressWarnings("squid:S3655")
+    public int execute()
+    {
+        final Time timer = Time.now();
+        final DatabaseConnection database = new DatabaseConnection(
+                this.optionAndArgumentDelegate.getOptionArgument(DATABASE_URL_INPUT).get());
+        final Connection databaseConnection = database.getConnection();
+        final String inputPath = this.optionAndArgumentDelegate.getOptionArgument(FLAG_PATH_INPUT)
+                .get();
+        this.timestamp = new Timestamp(Instant.now().toEpochMilli());
+
+        new File(inputPath).listFilesRecursively().forEach(file ->
+        {
+            // If this file is something we handle, read and upload the tasks contained within
+            final Optional<OutputFileType> optionalHandledFileType = getOptionalOutputType(file);
+            optionalHandledFileType.ifPresent(outputFileType ->
+            {
+
+                try (BufferedReader reader = this.getReader(file, outputFileType))
+                {
+
+                    final PreparedStatement flagSqlStatement = databaseConnection.prepareStatement(
```

Comment:
You should put your prepared statement in the try enclosure above so that they are automatically closed.